include/EncoderDecoder/Huffman/HuffmanEncoderDecoder.h

```diff
@@ -0,0 +1,101 @@
+#ifndef HUFFMAN_ENCODER_DECODER
+#define HUFFMAN_ENCODER_DECODER
+
+#include "BiMap/BiMap.h"
+#include "EncoderDecoder/AbstractEncoderDecoder.h"
+#include "EncoderDecoder/Huffman/PaddedString.h"
+#include "Pair/Hash.h"
+
+#include <algorithm>
+#include <bits/stdc++.h>
+#include <numeric>
+#include <vector>
+
+class HuffmanEncoderDecoder
+	: public AbstractEncoderDecoder<
+		  BiMap<char, std::pair<unsigned char, unsigned int>>,
+		  PaddedString> {
+private:
+	class Node {
+	public:
+		char character;
+		int frequency;
+		std::unique_ptr<Node> leftSon;
+		std::unique_ptr<Node> rightSon;
+
+		Node();
+		Node(char c);
+		Node(Node &&rhs) = default;
+		Node &operator=(Node &&rhs) = default;
+		bool operator==(const Node &rhs) const;
+		Node &operator++(int);
+	};
+
+public:
+	using MappingType = BiMap<char, std::pair<unsigned char, unsigned int>>;
+	using EncodedType = PaddedString;
+	using NodePtr = std::unique_ptr<Node>;
+	using NodeVector = std::vector<NodePtr>;
+	friend class HuffmanEncoderDecoder_MakeEncodingFromTextSubRoutines_Test;
+
+private:
+	static constexpr int MAX_ENCODED_BITS = 32;
```

Comment:
Why 32?

Reply:
This is part of the requirements.
The `.dane` file stores encodings in a way that 32 bits is the max encoding size.
Currently this affects the `MappingType` which maps a `char` to a `std::pair<unsigned char, unsigned int>` (`pair.first` is the size, and `pair.second` is the encoded character).
If I were to disregard this, I might get an integer overflow when creating a mapping.